modelmapper/s3.py

```diff
@@ -10,31 +13,75 @@ def Config(*args, **kwargs):
         raise NotImplementedError('Please install Boto3')
 
 
-class S3Mixin:
+class S3AccessError(ValueError):
+    pass
 
-    def verify_access_to_backup_source(self):
-        self._verify_access_to_s3_bucket()
 
-    def _verify_access_to_s3_bucket(self):
+class S3Base:
 
+    def _verify_access_to_s3_bucket(self, bucket):
+        try:
+            self.s3_client.list_objects_v2(Bucket=bucket, MaxKeys=1)
+        except Exception as e:
+            raise S3AccessError(f'Error when trying to access {bucket}: {e}')
+
+    @cached_property
+    def s3_client(self):
         config = Config(connect_timeout=.5, retries={'max_attempts': 1})
+        return boto3.client('s3', config=config)
 
-        s3_client = boto3.client('s3', config=config)
-        s3_client.list_objects_v2(Bucket=self.BUCKET_NAME, MaxKeys=1)
+    def get_list_of_files_on_s3(self, bucket, prefix='', max_keys=1000):
```

Comment:
Just as a warning this will return a max of 1000 keys even if you specify more (from the boto docs). You will want to look into `NextToken` for requests so you can page files correctly if you need to.